ipc/codegen/src/lib.rs

```diff
@@ -95,3 +95,58 @@ pub fn register(reg: &mut rustc_plugin::Registry) {
 
 	reg.register_attribute("ipc".to_owned(), AttributeType::Normal);
 }
+
+#[derive(Debug)]
+pub enum Error { InvalidFileName, ExpandFailure }
+
+pub fn derive_ipc(src_path: &str) -> Result<(), Error> {
+	use std::env;
+	use std::path::{Path, PathBuf};
+
+	let out_dir = env::var_os("OUT_DIR").unwrap();
+	let file_name = try!(PathBuf::from(src_path).file_name().ok_or(Error::InvalidFileName).map(|val| val.to_str().unwrap().to_owned()));
+
+	let mut intermediate_file_name = file_name.clone();
+	intermediate_file_name.push_str("rpc.in");
+
+	let intermediate_path = Path::new(&out_dir).join(&intermediate_file_name);
+	let final_path = Path::new(&out_dir).join(&file_name);
+
+	{
+		let mut registry = syntex::Registry::new();
+		register(&mut registry);
+		if let Err(_) = registry.expand("", &Path::new(src_path), &intermediate_path) {
+			// will be reported by compiler
+			return Err(Error::ExpandFailure)
```

Comment:
i think syntex now prints diagnostics in the stderr, so this error does not matter
like this

```bash
--- stderr
with_attrs.rs.in:32:41: 32:41 error: this file contains an un-closed delimiter
with_attrs.rs.in:32 impl IpcConfig for BadlyNamedService {}
                                                            ^
with_attrs.rs.in:26:24: 26:25 help: did you mean to close this delimiter?
with_attrs.rs.in:26 impl BadlyNamedService {
```